src/hotspot/share/opto/subnode.cpp

```diff
@@ -185,6 +189,18 @@ Node *SubINode::Ideal(PhaseGVN *phase, bool can_reshape){
       return new AddINode(in1, phase->intcon(-i->get_con()));
   }
 
+  // Convert "c0 - (x + c1)" into "(c0 - c1) - x"
+  if (in2->Opcode() == Op_AddI
+      && phase->type(in1)->isa_int() != NULL
+      && phase->type(in1)->isa_int()->is_con()
+      && phase->type(in2->in(2))->isa_int() != NULL
+      && phase->type(in2->in(2))->isa_int()->is_con()
+      && ok_to_convert(in2)) {
```

Comment:
`ok_to_convert` should go right after `in2->Opcode() == Op_AddI` for uniformity. Also you could reuse the method `ok_to_convert(Node*, Node*)` for this, just feed `in1` as the second argument similar to the check in line 218.